ext/gd/libgd/gd_gif_in.c

```diff
@@ -159,11 +159,7 @@ gdImagePtr gdImageCreateFromGifCtx(gdIOCtxPtr fd) /* {{{ */
 		return 0;
 	}
 
-	if (memcmp((char *)buf+3, "87a", 3) == 0) {
-		gif87a = 1;
-	} else if (memcmp((char *)buf+3, "89a", 3) == 0) {
-		gif87a = 0;
-	} else {
+	if (memcmp((char *)buf+3, "87a", 3) != 0 && memcmp((char *)buf+3, "89a", 3) != 0) {
```

Comment:
These changes should maybe also be merged upstream to libgd to keep the bundled lib in better sync

Reply:
Ok, I'll take care of it

Reply:
I see it has already been fixed (6 years ago) but [the fix is slightly different](https://github.com/libgd/libgd/commit/535eab566b4a412eef21347653fb98a48857d169#diff-61cc1502d11feef117a045a400fc5d2dR140).
Should I mimic the way they fix this?

Reply:
Yes, please. Actually, we should strive to sync our bundled libgd to the upstream lib, so we can eventually unbundle it.

Reply:
Done